src/fpm/generators/occ_grid.py

```diff
@@ -107,13 +104,17 @@ def generate_occ_grid(g, output_path, **custom_args):
 
 
 def draw_floorplan_obstacle(g, element, draw, west, south, fill, coords_map, **kwargs):
-    column_points = get_wall_points(g, element)
+    column_points = get_element_points(g, element)
     c_points = list()
+
+    laser_height = kwargs.get("map_laser_height", 0.7)
     for s in column_points:
-        c_coords = list()
-        for p in s.get("points"):
-            x, y, _ = get_waypoint_coord(g, p, coords_map)
-            c_coords.append([x, y, 0, 1])
+        height = s.get("height")
+        if laser_height > height:
+            # Don't process elements that are below the laser height
```

Comment:
Why only below and not above? Shouldn't it be a range around the height?

Reply:
Ah I didn't know about the assumption, I guess then you can't have objects on top of each other, e.g. cup on a table? I guess separate concepts then depend on what constitutes an "object"? Walls, columns & dividers are objects at the moment?

Reply:
There is no `Object` concept in the DSL at the moment. Most examples only include Walls/Columns/Dividers (since the focus is on architectural elements of a building). I guess, to do more complex shapes, e.g. to represent columns with base, shaft, etc. (see image), we would need this. 

![](https://www.designingbuildings.co.uk/wiki/File:Columnstructure2.jpg)

On the current examples I don't believe I have seen any elements being placed on top on one another. I suppose that would be an extension if we want to support it, but I'd only add this if we have a use case for it.

Reply:
I think it's fine leaving like it is right now, just document it somewhere perhaps (or even just clarify in comment) since it's not immediately obvious to me.

Reply:
The only other reasonable use of "suspended" wall-like elements I could think of is modelling [trims](https://www.riversidemillwork.ca/resources/blog/trim-and-moulding-terms/), e.g. window stools/aprons, casings or mouldings.